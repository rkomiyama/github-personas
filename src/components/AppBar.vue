<template>
  <v-app-bar
    height=64
    app
  >
    <v-app-bar-nav-icon
      class="hidden-md-and-up"
      @click.stop="$emit('click:drawer')"
    ></v-app-bar-nav-icon>
    <v-toolbar-title class="headline text-uppercase hidden-xs-only">
      <v-flex class="d-flex align-center">
        <v-img
          class="hidden-sm-and-down"
          :src="require('../assets/github.svg')"
          height=64
        />
        <span>GitHub&nbsp;</span>
        <span class="font-weight-light">PERSONAS</span>
      </v-flex>
    </v-toolbar-title>
    <v-spacer class="hidden-xs-only"></v-spacer>
    <v-text-field
      v-model="searchField"
      :label="searchFieldLabels[searchUserOption]"
      @blur="checkFieldChange"
      @change="changeUser"
      prepend-icon="mdi-account-search"
      hide-details
    >
    </v-text-field>
  </v-app-bar>
</template>

<script>
export default {
  name: 'AppBar',
  methods: {
    checkFieldChange () {
      if (this.oldSearchField === this.searchField) {
        this.userChange = false
      }
    },
    changeUser () {
      if (this.userChange) {
        this.$emit('change:user', this.searchField)
        this.oldSearchField = this.searchField
      } else {
        this.turnUserChangeOn()
      }
    },
    turnUserChangeOn () {
      this.userChange = true
    }
  },
  data () {
    return {
      oldSearchField: null,
      searchField: "",
      userChange: true,
      searchFieldLabels: {
        username: 'Search by Username',
        fullname: 'Search by Full Name'
      }
    }
  },
  props: {
    searchUserOption: String
  }
}
</script>
