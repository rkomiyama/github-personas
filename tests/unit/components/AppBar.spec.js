import Vue from 'vue';
import Vuetify from 'vuetify';
import { shallowMount } from '@vue/test-utils'
import AppBar from '@/components/AppBar.vue'

Vue.use(Vuetify)

describe('AppBar.vue', () => {
  it('is a Vue instance', () => {
    const wrapper = shallowMount(AppBar)
    expect(wrapper.isVueInstance()).toBeTruthy();
  })
})
