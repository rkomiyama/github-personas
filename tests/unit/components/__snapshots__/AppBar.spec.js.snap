// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar.vue AppBar renders the same HTML 1`] = `
<v-app-bar-stub height="64" tag="header" tile="true" extensionHeight="48" src="" value="true" app="true">
  <v-app-bar-nav-icon-stub class="hidden-md-and-up"></v-app-bar-nav-icon-stub>
  <v-toolbar-title-stub class="headline text-uppercase hidden-xs-only">
    <v-flex-stub tag="div" class="d-flex align-center">
      <v-img-stub height="64" position="center center" src="" transition="fade-transition" class="hidden-sm-and-down"></v-img-stub> <span>GitHubÂ </span> <span class="font-weight-light">PERSONAS</span>
    </v-flex-stub>
  </v-toolbar-title-stub>
  <v-spacer-stub class="hidden-xs-only"></v-spacer-stub>
  <v-text-field-stub errorCount="1" errorMessages="" messages="" rules="val =&gt; {
            /* istanbul ignore next */cov_bn3qa1xeo.f[1]++;

            const regex = /* istanbul ignore next */(cov_bn3qa1xeo.s[11]++, /^(([A-Za-z0-9])+(([A-Za-z0-9-])+([A-Za-z0-9]))?)*$/);
            /* istanbul ignore next */cov_bn3qa1xeo.s[12]++;
            return (/* istanbul ignore next */(cov_bn3qa1xeo.b[1][0]++, regex.test(val)) || /* istanbul ignore next */(cov_bn3qa1xeo.b[1][1]++, 'Invalid username.')
            );
          }" successMessages="" value="" backgroundColor="" hideDetails="true" label="Search by Username" prependIcon="mdi-account-search" loaderHeight="2" clearIcon="$vuetify.icons.clear" type="input" maxlength="39" data-test="searchFieldTest"></v-text-field-stub>
</v-app-bar-stub>
`;
